### PR TITLE
refactor: Move common functions to catchup-number file

### PR DESCRIPTION
**Pull request by SirusCodes**

### CODE REVIEW

**Comment by HarshKapadia2** [comment]
Good optimizations, @SirusCodes! Thank you!
There are some changes and I'll let you know those through reviews.

File: util/catchup-number.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    if (lastChar === "1") return "st";
+    else if (lastChar === "2") return "nd";
+    else if (lastChar === "3") return "rd";
+    else return "th";
+}
+
+async function getCatchupNumber(getNextCatchupNumber, addSuffix) {
+    catchUpNumbers = await getCatchUpNumber();
+    let catchUpNumber = catchUpNumbers[0];
```
Comment (HarshKapadia2) [inline util/catchup-number.js:31]: Variable and function names are super confusing here. Let us change it up.
Please make corresponding changes in this file and the other two files.

- Rename `getCatchUpNumber` to `getAllCatchUpNumbers`
- Rename `getCatchupNumber` to `getCatchUpNumber`
- Rename `catchUpNumbers` to `allCatchUpNumbers`

File: util/catchup-number.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        const suffix = getSuffix(catchUpNumber);
+        catchUpNumber = catchUpNumber + suffix;
+    }
+
+    return catchUpNumber;
+}
+
+module.exports = {
+    getCatchupNumber,
+};
```
Comment (HarshKapadia2) [inline util/catchup-number.js:48]: Please add a new line at the end of all files.

File: util/get-latest-catchup-number.js [hunk 3]
```diff
@@ -1,26 +1,5 @@
-const { readdir } = require("fs/promises");
+const catchupNumber = require('./catchup-number');
```
Comment (HarshKapadia2) [inline util/get-latest-catchup-number.js:1]: Please use double quotes instead of single quotes to stay consistent. Please change in all files.

Comment (HarshKapadia2) [reply util/get-latest-catchup-number.js:1]: Please use [Prettier](https://prettier.io) to format your code.

File: util/get-latest-catchup-number.js [hunk 4]
```diff
@@ -1,26 +1,5 @@
...
-		dirArray[i] = parseInt(dirArray[i]);
-	return dirArray.reverse();
-}
-
-async function getDirectoryNames(path) {
-	return (await readdir(path, { withFileTypes: true }))
-		.filter((dirent) => dirent.isDirectory())
-		.map((dirent) => dirent.name);
-}
+catchupNumber.getCatchupNumber(false, false).then(catchUpNumber => {
```
Comment (HarshKapadia2) [inline util/get-latest-catchup-number.js:3]: Add a `.catch()` to handle errors using `console.error()`.

Eg:
```javascript
fn.then(...).catch((err) => console.error(err));
```

Comment (HarshKapadia2) [reply util/get-latest-catchup-number.js:3]: Please add in all files.

**Review by HarshKapadia2 — CHANGES_REQUESTED** [review]

File: util/catchup-number.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+const { readdir } = require("fs/promises");
+
+const path = __dirname + "/../summary/sessions";
+
+async function getCatchUpAllNumbers() {
```
Comment (HarshKapadia2) [inline util/catchup-number.js:5]: The function name should be `getAllCatchUpNumbers`. Please make corresponding changes.

**Review by HarshKapadia2 — CHANGES_REQUESTED** [review]

File: util/get-next-catchup-number.js [hunk 6]
```diff
@@ -1,38 +1,7 @@
...
-	if (lastChar === "1") return "st";
-	else if (lastChar === "2") return "nd";
-	else if (lastChar === "3") return "rd";
-	else return "th";
-}
+catchupNumber
+    .getCatchupNumber(true, true)
+    .then(console.log)
+    .catch(console.error);
+
```
Comment (HarshKapadia2) [inline util/get-next-catchup-number.js:7]: There seems to be a line ending issue here. Please check.

**Review by HarshKapadia2 — CHANGES_REQUESTED** [review]

File: util/catchup-number.js (hunk 5 above)
Comment (SirusCodes) [reply util/catchup-number.js:5]: damn😂

**Review by HarshKapadia2 — APPROVED** [review]

**Comment by HarshKapadia2** [comment]
LGTM! Let's hope the workflows run. I'll be checking that after merging.

**Comment by SirusCodes** [comment]
Prefect!!

**Pull request merged**